lib/pbench/cli/agent/commands/generate_token.py

```diff
@@ -0,0 +1,71 @@
+"""pbench-generate-token"""
+
+import click
+import json
+import requests
+
+from pbench.agent.base import BaseCommand
+from pbench.cli.agent import pass_cli_context
+from pbench.cli.agent.options import common_options
+
+
+class GenerateToken(BaseCommand):
+    def __init__(self, context):
+        super().__init__(context)
+
+    def execute(self):
+        """Generate a token used to access the pbench server RESTful API"""
+
+        # FIXME:  Should we be logging anything here?
+        #
+        #  self.logger.warn(
+        #     "%s deprecated, will be removed in future release in favor of pbench-clear-results",
```

Comment:
Seems unnecessary?

Reply:
Funny, but the whole commented logger entry from a completely different context bothers me more than the thought of whether we should log here.

I agree with Pete; we should probably log any errors or unusual behavior, but I'm not sure I see any need to log every attempt at generating a PUT token.

Reply:
Yeah, line 22 is just a remnant from my cookie-cuttering this module from something else.  I would have removed it, but I thought that we might be wanting to log _something_...maybe.

As the code stands, I don't think it currently detects "unusual behavior".  I'll grant that maybe we don't want to log successful logins (that's an audit function that we aren't currently pursuing).  Do we actually want to log failures in the tool?

So, I'm thinking there is nothing to log here.  I'm thinking that any logging would/should be done in the server, instead.  Anyone disagree?

Reply:
The function here is pretty simple. While I did add some logging in my `pbench-state-manager.py` tool (which is mostly for the unit tests), the way I handled errors was to simply write the `str()` of any exception to `stderr` as a failure message and exit with `1`. That might well be good enough here, too, even though this is more user-facing.